lang_python/parsing/AST_python.ml

```diff
@@ -319,6 +319,8 @@ and param_pattern =
 *)
 and type_ = expr
 
+and lsif_type = Type of type_ | Arrow of parameters * type_
```

Comment:
this requires a comment (it's ok to mention deep-semgrep or semgrep-proprietary)